code/game/gamemodes/blob/blobs/blob_mobs.dm

```diff
@@ -244,6 +245,10 @@
 			if(overmind)
 				I.color = overmind.blob_reagent_datum.complementary_color
 			flick_overlay_view(I, src, 8)
+		if(factory && (locate(/obj/structure/blob/core) in blobs_in_area))
+			adjustHealth(-maxHealth*0.1)
+		if(factory && (locate(/obj/structure/blob/node) in blobs_in_area))
+			adjustHealth(-maxHealth*0.05)
```

Comment:
These 4 lines should be tabbed in once more because damagesources is guaranteed to be > 0 if it runs